### ISSUE TITLE
Connecting to broker now blocks forever by default

### ISSUE THREAD

**Issue by matusvalo** [celery/kombu#1210]
After #1193 several methods started to block connection to broker till is not established. This can take forever when  broker won't appear. Moreover, it changed raised exceptions to OperationalError instead of native transport one. It seems that this change affects users  - see: #429 

The semantics of this should be at least postponed an introduced properly after proper discussion.

**Comment by auvipy** [comment]
thanks for digging this.

**Comment by matusvalo** [comment]
No problem @auvipy . My appologies for unexpected issues :-/.
